fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/HyperGraphComparator.java

```diff
@@ -108,16 +109,22 @@ private ComparisonResult isLogicCompatible() {
     private ComparisonResult buildComparisonRes() {
         ComparisonResult.Builder builder = new ComparisonResult.Builder();
         for (Entry<Edge, List<? extends Expression>> e : pullUpQueryExprWithEdge.entrySet()) {
-            if (!e.getValue().isEmpty() && !canPullUp(e.getKey())) {
+            List<? extends Expression> rawFilter = e.getValue().stream()
+                    .filter(expr -> !expr.isInferred())
```

Comment:
this should use method `org.apache.doris.nereids.util.ExpressionUtils#isInferred`